derive-macros/src/lib.rs

```diff
@@ -45,10 +45,10 @@ pub fn derive_schema(input: proc_macro::TokenStream) -> proc_macro::TokenStream
     let schema_fields = gen_schema_fields(&input.data);
     let output = quote! {
         #[automatically_derived]
-        impl crate::actions::schemas::ToDataType for #struct_ident {
-            fn to_data_type() -> crate::schema::DataType {
+        impl crate::actions::schemas::ToSchema for #struct_ident {
+            fn to_schema() -> crate::schema::StructType {
```

Comment:
> > Updated the schema derivation code to include a new ToSchema trait. This eliminates the need for various fallible project operations in the code.
> 
> is the new `ToSchema` just a rename of `ToDataType` though? (just making sure i'm understanding correctly)

No, we used do return `DataType::Struct` and now we produce the inner `StructType` directly. And then there's a blanket impl for `ToDataType` that invokes `T::to_schema().into()`.

Reply:
Ah i see, so we have both now - thanks!